superchain/superchain.go

```diff
@@ -106,9 +107,6 @@ type ChainConfig struct {
 	// Superchain is a simple string to identify the superchain.
 	// This is implied by directory structure, and not encoded in the config file itself.
 	Superchain string `yaml:"-"`
-	// Chain is a simple string to identify the chain, within its superchain context.
-	// This matches the resource filename, it is not encoded in the config file itself.
-	Chain string `yaml:"-"`
```

Comment:
Removing this is a compile-time breaking change for
* op-geth
* op-node
which, even when patched, counts as a runtime breaking change for the `-network` and `-op-network` flags when starting these programs.

The shortest path to resolving this would be to keep the `Chain` property, and keep the code that sets it to the filename.

* New chains can have `Chain` = `ShortName` = file name
* existing chains keep `Chain` = file name, and get a new `ShortName` which isn't used for anything other than global chain id check
* we deprecate `Chain` and at some future point remove it and fix the downstream problems that will create.